wrappers/javascript/aries-askar-shared/src/store/StoreKeyMethod.ts

```diff
@@ -0,0 +1,42 @@
+import { AriesAskarError } from '../error'
+
+export enum KdfMethod {
+  Raw = 'raw',
+  Kdf = 'kdf',
+  None = 'none',
+}
+
+export enum Argon2Level {
+  Interactive = 'int',
+  Moderate = 'mod',
+}
+
+export class StoreKeyMethod {
+  private method: KdfMethod
+  private argon2Level?: Argon2Level
+
+  public constructor(method: KdfMethod, argon2Level?: Argon2Level) {
+    if (method == KdfMethod.Kdf && !argon2Level) {
```

Comment:
```suggestion
    if (method === KdfMethod.Kdf && !argon2Level) {
```

Reply:
Should we add a check if kdf level is Raw and argon2level is defined? Also should we just add more enum values instead? What's the benefit of having kdf and then mod/int instaed of having kdf_mod and kdf_int?

Reply:
Mainly did it to be inline with how it is done in askar itself and this allows, although not quite good, more kdfs.

Reply:
I will change it to `Argon2IMod` and `Argon2IInt` as growing with more kdfs would also not be a breaking change then.